core/src/main/java/io/questdb/griffin/FunctionParser.java

```diff
@@ -459,14 +460,41 @@ private Function createConstant(int position, final CharSequence tok) throws Sql
             return Constants.getTypeConstant(columnType);
         }
 
-        // geohash?
+        // geohash type constant
+
         if (startsWithGeoHashKeyword(tok)) {
-            int bits = SqlParser.parseGeoHashSize(position, 7, tok);
-            return GeoHashTypeConstant.getInstanceByPrecision(bits);
+            return GeoHashTypeConstant.getInstanceByPrecision(
+                    SqlParser.parseGeoHashSize(position, 7, tok));
+        }
+
+        if (len > 1 && tok.charAt(0) == 35) { // '#'
```

Comment:
How `'#1234'` will parse? Should be a string

Reply:
it is a geohash chars constant, same as `cast('1234' as geohash(4c))`

Reply:
How to create string starting with # in query? For example I want to query 

```
create table as (select rnd_str('#1234', '#88484') from long_sequence(1000) );

select * from x where str = '#1234';
```

Will this work without errors and return half of the records?

Reply:
if the parser sees unquoted #1234 then it is a geohash, because '#' flags the token as a geohash chars constant.
the FunctionParser.createConstant gets this token and parses it `GeoHash.fromString("1234"...` .
if the parser sees quoted '#1234' then it is a string. It could be a geohash with `cast('#1234') as geohash(4c))`.